pyvcloud/vcd/vdc.py

```diff
@@ -373,6 +373,42 @@ def list_edge_gateways(self):
                 })
         return edge_gateways
 
+    def get_nat_rules(self, edge_gateway_name):
+        """Request the nat rules of an edge gateway.
+
+        :param edge_gateway_name: (str): The name of the edge gateway.
+
+        :return: A list of :class:`lxml.objectify.StringElement' objects
+            representing the nat rules.
+
+        :raises: Exception: if there is no edge gateway with the provided name.
+        :raises: Exception: if there are no nat rules on the edge gateway
+            with the provided name.
+        """
+        edge_gateways = self.list_edge_gateways()
+        nat_rules = []
+        for edge_gateway in edge_gateways:
```

Comment:
This piece of code that retrieves the full representation of an edge gateway identified by name, should be refactored out in a separate function. I am sure that this functionality will be a recurring theme while accessing/updating other services/properties of an edge gateway.